src/Builder.ts

```diff
@@ -1,5 +1,5 @@
 export type IBuilder<T> = {
-  [k in keyof T]-?: ((arg: T[k]) => IBuilder<T>) & (() => T[k]);
+  [k in keyof T]-?: ((arg: Exclude<T[k], undefined>) => IBuilder<T>) & (() => T[k]);
```

Comment:
@Vincent-Pang 

I really don't know why we can see getters in builder, but may be need to change getter return type to `(() => T[k] | undefined)`

?